docs/sources/shared/configuration.md

```diff
@@ -4919,6 +4919,13 @@ engine:
   # CLI flag: -querier.engine.batch-size
   [batch_size: <int> | default = 100]
 
+  # Experimental: The number of inputs that are prefetched simultaneously by any
+  # Merge node. A value of 0 means that only the currently processed input is
+  # prefetched, 1 means that only the next input is prefetched, and so on. A
+  # negative value means that all inputs are be prefetched in parallel.
+  # CLI flag: -querier.engine.merge-prefetch-concurrency
+  [merge_prefetch_concurrency: <int> | default = 1]
```

Comment:
I think the `_concurrency` suffix makes this a little misleading. A value of 1 to me reads there's at most one concurrent worker (i.e., sequential reads).

Is there a different name we can use here? Off the top of my head, a few options might be `merge_prefetch_inputs`, `merge_num_inputs_prefetch`, or maybe `merge_prefetch_count`.